Sources/SwiftFusion/Core/FixedSizeMatrix.swift

```diff
@@ -325,7 +322,7 @@ extension FixedSizeMatrix {
 @differentiable
 public func matvec<Rows>(_ lhs: FixedSizeMatrix<Rows>, _ rhs: Rows.Element) -> Rows.Element {
   precondition(type(of: lhs).isSquare, "matvec only implemented for square matrices")
-  var r = Rows.Element.zero
+  var r = Rows.Element(repeatElement(0.0, count: rhs.dimension))
```

Comment:
Using `repeatElement(0.0, count: rhs.dimension)` won't work to initialize zero vectors of all types (like the dictionary-like heterogeneous vector arrays).

Those will need something like `.init(zerosCompatibleWith:)`.

But this PR is already an improvement so we could make that improvement later in a separate PR.

(This is similar to my comment in https://github.com/borglab/SwiftFusion/pull/176#discussion_r474169145).

Reply:
Yes, one step at a time please.

Reply:
Also, how is such a vector going to be the row of a `FixedSizeMatrix`?